contracts/core/libs/sismo-gated/SismoGated.sol

```diff
@@ -0,0 +1,210 @@
+  address public constant BADGES_LOCAL_ADDRESS = 0xeF5b2Be9a6075a61bCA4384abc375485d5e196c3;
+  address public constant HYDRA_S1_ACCOUNTBOUND_LOCAL_ADDRESS =
+    0xf93A0C43A3466488D416628bf149495285e9f274;
+
+  Badges public immutable BADGES;
+  HydraS1AccountboundAttester public immutable HYDRA_S1_ACCOUNTBOUND_ATTESTER;
+
+  error UnsupportedNetwork();
+  error InvalidArgumentsLength();
+  error UserIsNotOwnerOfBadge(uint256 badgeTokenId, uint256 balance);
+  error AccountAndRequestDestinationDoNotMatch(address account, address requestDestination);
+
+  /**
+   * @dev Constructor
+   */
+  constructor() {
+    // select the correct contract addresses based on the network
+    (address badgesAddress, address hydraS1AccountboundAttesterAddress) = _getContractAddresses();
+    BADGES = Badges(badgesAddress);
+    HYDRA_S1_ACCOUNTBOUND_ATTESTER = HydraS1AccountboundAttester(
+      hydraS1AccountboundAttesterAddress
+    );
+  }
+
+  /**
+   * @dev Modifier allowing only:
+   *      - Owners of the badge with tokenId == badgeTokenId and a badge balance >= minBalance
+   *      - Addresses used in a valid proof of elibility that can be provided in `sismoProofData`
+   * @param account Address which holds the badge or is eligible for the badge (need to provide the correct proof for this account)
+   * @param badgeTokenId Token ID of the badge that the user needs to hold to access the function
+   * @param minBalance Minimum balance (= level) of the badge that the user needs to hold to access the function (if no proof is provided)
+   * @param attester Attester contract used to verify the proof
+   * @param sismoProofData Data containing the user request and the proof of eligibility associated to it
+   */
+  modifier onlyBadgeOwnerOrValidProof(
+    address account,
+    uint256 badgeTokenId,
+    uint256 minBalance,
+    Attester attester,
+    bytes memory sismoProofData
+  ) {
+    uint256[] memory badgeTokenIds = new uint256[](1);
+    badgeTokenIds[0] = badgeTokenId;
+
+    uint256[] memory minBalances = new uint256[](1);
+    minBalances[0] = minBalance;
+
+    Attester[] memory attesters = new Attester[](1);
+    attesters[0] = attester;
+
+    bytes[] memory sismoProofDataArray = new bytes[](1);
+    sismoProofDataArray[0] = sismoProofData;
+
+    checkAccountBadgesOrSismoProofs(
+      account,
+      badgeTokenIds,
+      minBalances,
+      attesters,
+      sismoProofDataArray
+    );
+
+    _;
+  }
+
+  /**
+   * @dev Check if the `account` address holds badges or is used in valid sismo proofs
+   * @param account Address which holds badges or is used in valid sismo proofs
+   * @param badgeTokenIds Token ID of the badges
+   * @param minBalances Minimum balances (= levels) of the badges
+   * @param attesters Attester contracts used to verify proofs
+   * @param sismoProofDataArray Array of bytes containing the user requests and the proofs of each badge eligibility
+   */
+  function checkAccountBadgesOrSismoProofs(
```

Comment:
I think it still does not check min value with proof with sismo.

I can generate an attestation of level 0